### PR TITLE
feat(DRS): add a new resources to manage DRS

### LINKED ISSUE #1950: [DRS] Add DRS service support
I'd like to add DRS service support on terraform side.

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
1. add a new resource to manage DRS job

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1950 

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/drs' TESTARGS='-run=TestAccResourceDrsJob_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/drs -v -run=TestAccResourceDrsJob_basic -timeout 360m -parallel 4
=== RUN   TestAccResourceDrsJob_basic
=== PAUSE TestAccResourceDrsJob_basic
=== CONT  TestAccResourceDrsJob_basic
--- PASS: TestAccResourceDrsJob_basic (1316.28s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/drs       1316.421s
```

### CODE REVIEW

**Comment by niuzhenguo** [comment]
need to update golangsdk, already merged.

File: docs/resources/drs_job.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+* `ssl_cert_key` - (Optional, String, ForceNew) Specifies the SSL certificate content, encrypted with base64.
+ It is mandatory when `ssl_enabled` is `true`. Changing this parameter will create a new resource.
+
+* `ssl_cert_name` - (Optional, String, ForceNew) Specifies SSL certificate name.
+ It is mandatory when `ssl_enabled` is `true`. Changing this parameter will create a new resource.
+
+* `ssl_cert_check_sum` - (Optional, String, ForceNew) Specifies the checksum of SSL certificate content.
+ It is mandatory when `ssl_enabled` is `true`. Changing this parameter will create a new resource.
+
+* `ssl_cert_password` - (Optional, String, ForceNew) Specifies SSL certificate password. It is mandatory when
```
Comment (Lance52259) [inline docs/resources/drs_job.md:192]: Has the second half of this parameter description been accidentally deleted?

File: docs/resources/drs_job.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` -  The resource ID in UUID format.
+
+* `created_at` - Create time. The format is ISO8601:YYYY-MM-DDThh:mm:ssZ
+
+* `status` - Status.
+
+* `publid_ip` - Public ip.
```
Comment (Lance52259) [inline docs/resources/drs_job.md]: `IP`

File: docs/resources/drs_job.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+ It is mandatory when `ssl_enabled` is `true`. Changing this parameter will create a new resource.
+
+* `ssl_cert_check_sum` - (Optional, String, ForceNew) Specifies the checksum of SSL certificate content.
+ It is mandatory when `ssl_enabled` is `true`. Changing this parameter will create a new resource.
+
+* `ssl_cert_password` - (Optional, String, ForceNew) Specifies SSL certificate password. It is mandatory when
+
+The `limit_speed` block supports:
+
+* `speed` - (Required, String, ForceNew) Specifies the transmission speed, the value range is 1 to 9999,unit: `MB/s`.
```
Comment (Lance52259) [inline docs/resources/drs_job.md]: Missing a blank space between `9999,` and `unit`.

File: docs/resources/drs_job.md [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+  + **cloudDataGuard-mysql**: Disaster recovery use.
+  + **gaussdbv5**: GaussDB (for openGauss) synchronization use.
+
+* `direction` - (Required, String, ForceNew) Specifies the direction of data flow.
+ Changing this parameter will create a new resource. The options are as follows:
+  + **up**: To the cloud. The destination database must be a database in the current cloud.
+  + **down**: Out of the cloud. The source database must be a database in the current cloud.
+  + **non-dbs**: self-built database.
+  
+* `source_db` - (Required, List,ForceNew) Specifies the source database configuration.
```
Comment (Lance52259) [inline docs/resources/drs_job.md]: Missing a blank space between `List` and `ForceNew`.
The same problems exists in this documentation.

File: huaweicloud/services/drs/resource_huaweicloud_drs_job.go [hunk 5]
```diff
@@ -0,0 +1,810 @@
...
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^([A-Za-z][A-Za-z0-9-_\\.]{3,49})$`),
```
Comment (Lance52259) [inline huaweicloud/services/drs/resource_huaweicloud_drs_job.go]: Is `A-Za-z0-9-_\.` or `A-Za-z0-9-_.` ?
Suggest split the length check as StringLenBetween(4, 50) to manage string format.

File: huaweicloud/services/drs/resource_huaweicloud_drs_job.go [hunk 6]
```diff
@@ -0,0 +1,810 @@
...
+
+	return &nodeResource
+}
+
+func resourceDrsJobCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.DrsV3Client(region)
+	if err != nil {
+		return diag.Errorf("error creating DRS v3 client, err=%s", err)
```
Comment (Lance52259) [inline huaweicloud/services/drs/resource_huaweicloud_drs_job.go]: This error return is the outermost output, and the first letter can be expressed in uppercase (the beginning of the sentence).

File: huaweicloud/services/drs/resource_huaweicloud_drs_job.go [hunk 7]
```diff
@@ -0,0 +1,810 @@
...
+			{
+				JobId:     jobId,
+				StartTime: d.Get("start_time").(string),
+			},
+		},
+	}
+	_, err = jobs.Start(client, startReq)
+
+	if err != nil {
+		return fmtp.DiagErrorf("start DRS job failed,error:%s", err)
```
Comment (Lance52259) [inline huaweicloud/services/drs/resource_huaweicloud_drs_job.go]: blank space: ditto
format: ditto

File: huaweicloud/config/endpoints.go [hunk 8]
```diff
@@ -251,6 +251,10 @@ var allServiceCatalog = map[string]ServiceCatalog{
 		Name:    "gaussdb-opengauss",
 		Version: "v3",
 	},
+	"drs": {
+		Name:    "drs",
+		Version: "v3",
+	},
```
Comment (Lance52259) [inline huaweicloud/config/endpoints.go:279]: After #1967 merged, please add `drs` to the map.

File: docs/resources/drs_job.md (hunk 1 above)
Comment (hmbig2) [reply docs/resources/drs_job.md:192]: fix

File: huaweicloud/services/drs/resource_huaweicloud_drs_job.go (hunk 5 above)
Comment (hmbig2) [reply huaweicloud/services/drs/resource_huaweicloud_drs_job.go]: fix

File: docs/resources/drs_job.md (hunk 3 above)
Comment (hmbig2) [reply docs/resources/drs_job.md]: fix

File: huaweicloud/services/drs/resource_huaweicloud_drs_job.go (hunk 6 above)
Comment (hmbig2) [reply huaweicloud/services/drs/resource_huaweicloud_drs_job.go]: fix

File: huaweicloud/config/endpoints.go (hunk 8 above)
Comment (hmbig2) [reply huaweicloud/config/endpoints.go:279]: no need

File: docs/resources/drs_job.md (hunk 2 above)
Comment (hmbig2) [reply docs/resources/drs_job.md]: fix

File: docs/resources/drs_job.md [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+subcategory: "Data Replication Service (DRS)"
+---
+
+# huaweicloud_drs_job
+
+Manages DRS job resource within HuaweiCloud.
+
+## Example Usage
+
+### Create a DRS job to migrate data to the huaweiCloud RDS database
```
Comment (ShiChangkuo) [inline docs/resources/drs_job.md]: huaweiCloud should be capitalized

File: huaweicloud/services/drs/resource_huaweicloud_drs_job.go [hunk 10]
```diff
@@ -0,0 +1,818 @@
...
+	jobId := rst.Results[0].Id
+
+	err = waitingforJobStatus(ctx, client, jobId, "create", d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(jobId)
+
+	valid := testConnections(client, jobId, opts.Jobs[0])
```
Comment (ShiChangkuo) [inline huaweicloud/services/drs/resource_huaweicloud_drs_job.go:337]: can we test connections before create the DRS job?

Comment (hmbig2) [reply huaweicloud/services/drs/resource_huaweicloud_drs_job.go:337]: NO, jobId is required

File: docs/resources/drs_job.md (hunk 9 above)
Comment (hmbig2) [reply docs/resources/drs_job.md]: fix

**Pull request merged**